lib/harvester.js

```diff
@@ -197,6 +200,29 @@ Harvester.prototype._preprocessSchema = function (schema) {
   return schema;
 };
 
+Harvester.prototype._convertToLegacySchema = function(schema) {
+  var legacySchema = {};
+  var schemaMap = {
+    'string' : String,
+    'number' : Number,
+    'date' : Date,
+    'buffer' : Buffer,
+    'boolean' : Boolean,
+    'array' : Array
+  };
+
+  _.each(schema, function(item, key) {
+    if(item.isJoi) {
+      legacySchema[key] = schemaMap[item._type];
+    } else {
+      legacySchema[key] = item;
```

Comment:
We're missing one more feature: handling the new 'links' parameter.  Line 216 won't work properly for that since it won't handle the fact that we previously didn't have a 'links' parameter. An easy way to achieve quick compliance is something like if (key=='links'){ iterate over item and assign those sub items to their sub keys on the legacy schema}